### PR TITLE
ARC-80 add hierarchy via env

### PR DESCRIPTION
**Pull request by tofixx**
# Description
This PR adds hierarchical parsing of env-values defined in options.loadFilesFromEnv instead of parsing NODE_ENV only.

Besides that, this ensures Changelog entries to be added.

## Approval for review

- [x] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.

> Notice: Please remove the WIP label if the PR is ready to review, otherwise nobody will review it.

### Link to Definiton of Done

More and detailed information on the _definition of done_ can be found [on Confluence](https://docs.schul-cloud.org/pages/viewpage.action?pageId=92831762)

### CODE REVIEW

File: src/configuration/index.ts [hunk 1]
```diff
@@ -33,6 +34,7 @@ export const defaultOptions: IRequiredConfigOptions = {
 	throwOnError: true,
 	allowRuntimeChangesInEnv: ['test'],
 	defaultNodeEnv: 'development',
+	loadFilesFromEnv: [NODE_ENV, 'SC_INSTANCE'],
```
Comment (CeEv) [inline src/configuration/index.ts:37]: The file ending .json and path is added later? What is mean with NODE_ENV='NODE_ENV'?

Maybe loadFilesFromEnv is confused. This are only the fileNames, that should loaded later

File: src/configuration/index.ts [hunk 2]
```diff
@@ -128,27 +130,57 @@ export class Configuration implements IConfiguration {
...
-			configurationFileNames.push(this.NODE_ENV + '.json');
+		// add environment files
+		if (
+			Array.isArray(this.options.loadFilesFromEnv) &&
+			this.options.loadFilesFromEnv.length != 0
+		) {
+			this.options.loadFilesFromEnv.forEach((envName) => {
+				let ok = true;
+				// add configuration based on current envName, ignore default already added first
+				if (!(envName in dotAndEnv)) {
```
Comment (CeEv) [inline src/configuration/index.ts:154]: if () {} else if() {} else {} and remove ok

File: src/configuration/index.ts [hunk 3]
```diff
@@ -128,27 +130,57 @@ export class Configuration implements IConfiguration {
...
+		const configurationFileNamesAddedSuccessfully = [];
+		const configurations = [];
+		// start with default file
 		configurationFileNames.push('default.json');
-		if (this.NODE_ENV !== 'default') {
-			configurationFileNames.push(this.NODE_ENV + '.json');
+		// add environment files
+		if (
+			Array.isArray(this.options.loadFilesFromEnv) &&
+			this.options.loadFilesFromEnv.length != 0
```
Comment (CeEv) [inline src/configuration/index.ts:149]: !==

File: src/configuration/index.ts [hunk 4]
```diff
@@ -128,27 +130,57 @@ export class Configuration implements IConfiguration {
...
+		logger.info(
+			'will parse following configuration filenames in given order',
+			configurationFileNames
+		);
 		for (const file of configurationFileNames) {
 			const fullFileName = path.join(
 				this.options.baseDir,
 				this.options.configDir,
-				file
+				file + '.json'
```
Comment (CeEv) [inline src/configuration/index.ts:175]: file,
'.json'

Comment (CeEv) [reply src/configuration/index.ts:175]: I am not sure what it is loaded in default case or with some settings. We should speak about it later the day.

**Review by CeEv — CHANGES_REQUESTED** [review]

File: src/configuration/index.ts (hunk 4 above)
Comment (tofixx) [reply src/configuration/index.ts:184]: Currently, JSON files are supported only. The extension was added before into configurationFileNames for every single entry which has been changed here.

**Review by CeEv — APPROVED** [review]

File: src/configuration/index.ts (hunk 4 above)
Comment (tofixx) [reply src/configuration/index.ts:182]: adding json as parameter would finally handle file as folder name.

**Pull request merged**